src/processout/processout.ts

```diff
@@ -864,6 +908,35 @@ module ProcessOut {
             }.bind(this);
         }
 
+        protected buildHandleTokenInvoiceAction(
+          invoiceID: string,
+          gatewayConf: any,
+          customerTokenId?: string,
+        ): (
+          tokenized: (token: string) => void,
+          tokenError: (err: Exception) => void
+        ) => ActionHandler {
+          if (!customerTokenId) {
+            return this.buildHandleInvoiceAction(invoiceID, gatewayConf);
+          }
+
+          return function (
+            tokenized: (token: string) => void,
+            tokenError: (err: Exception) => void,
+            iframeOverride?: IframeOverride,
+          ): ActionHandler {
+            return this.handleConfigurableInvoiceAction({
+              invoiceId: invoiceID,
+              tokenized: tokenized,
+              tokenError: tokenError,
+              gatewayConf: mapInvoiceGatewayConfigurationProps(gatewayConf),
+              customerTokenId: customerTokenId,
+              additionalData: {},
```

Comment:
Interesting, so `additionalData` was not passed here previously as well? Does it mean that merchant has no option to pass it in this flow or is it just handled earlier or something?

Reply:
Not via this route. But, according to our [official docs](https://docs.processout.com/docs/sdks-hosted-payment-page#list-the-available-apms-and-redirect) they could use the `client.handleInvoiceActionAdditionalData()` function directly.